src/realm/sync/transform.cpp

```diff
@@ -895,6 +892,25 @@ void TransformerImpl::MinorSide::prepend(InputIterator begin, InputIterator end)
 
 namespace {
 
+REALM_NORETURN void throw_bad_merge(std::string msg)
+{
+    throw sync::TransformError{std::move(msg)};
+}
+
+template <class... Params>
+REALM_NORETURN void bad_merge(const char* msg, Params&&... params)
+{
+    throw_bad_merge(util::format(msg, std::forward<Params>(params)...));
+}
+
+REALM_NORETURN void bad_merge(_impl::TransformerImpl::Side& side, Instruction::ObjectInstruction instr,
```

Comment:
Do we want to have a bad_merge for PathInstructions too and log which property/path was being touched?

Reply:
Done. It turns out that they were all `PathInstruction`'s anyway.